stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -437,6 +436,39 @@ function A_mul_B_td!(C::AbstractMatrix, A::BiTriSym, B::BiTriSym)
     C
 end
 
+function A_mul_B_td!(C::AbstractMatrix, A::BiTriSym, B::Diagonal)
```

Comment:
Similarly to the comment below: do we need methods for `Transpose`s and `Adjoint`s of `B`? In line 634, you call `A_mul_B_td!` with a `B::Transpose{<:Any, <:Diagonal}`.

Hm, actually, it seems that the exact same code works for `B::Union{Diagonal, Adjoint{<:Any, <:Diagonal}, Transpose{<:Any, <:Diagonal}`. The elementwise transposition/adjoint is done at `getindex`. You could define a constant `AdjOrTransDiagonal` somewhere above?